### DISCUSSION TITLE
Emit event in table action

### DISCUSSION THREAD

**Discussion by korematter** [filamentphp/filament#6268] [Help] [answered]
Hi,

I'm trying to emit a livewire event in table action 'addToCart' to update a shopping cart count in the header but can't seem to pass in the $livewire component to the 'after' callback. 

![image](https://user-images.githubusercontent.com/58536534/233348170-629b8fcf-e27c-43b9-89e4-bab2b866b336.png)

I get this error saying $livewire is ListEntries

App\Filament\Resources\EntryResource::App\Filament\Resources\{closure}(): Argument #1 ($livewire) must be of type Livewire\Livewire, App\Filament\Resources\EntryResource\Pages\ListEntries given, called in /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php on line 36

Here's the bit of code: 

```
->actions([
        Tables\Actions\ViewAction::make()
            ->visible(fn (Entry $record) => $record->status === Status::ENTRY_DRAFT ),
        Tables\Actions\EditAction::make()
            ->visible(fn (Entry $record) => ! ($record->status === Status::ENTRY_DRAFT) ),
        Tables\Actions\Action::make('addToCart')
            ->action(function (Entry $entry) {
                DB::table('entries')
                    ->where('id', $entry->id)
                    ->update(['status_id' => Status::ENTRY_INCART]);
            })
            ->after(fn(Livewire $livewire, Entry $entry) => $livewire->emit('refreshTopbarShoppingCartCount', $entry))
            ->color('success')
            ->icon('heroicon-s-shopping-cart'),
            ])
```

Is there a way to emit a livewire event in this action closure?

Thanks,
Steve

**Comment by danharrin** [comment, marked as answer]
You have imported the wrong class. Livewire components do not extend `Livewire\Livewire`, they extend `Livewire\Component`. You either need to alias this import, use `Component $livewire` or use `ListEntries $livewire`

**Comment by korematter** [comment]
Ah! Thanks Dan. I'm an idiot. Appreciate the quick response!
